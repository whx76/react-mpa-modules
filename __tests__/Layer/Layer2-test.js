import React from 'react';
import ReactDOM from 'react-dom';
import LayerComponent from '../../WrcLayer/lib/Layer2.jsx';
import renderer from 'react-test-renderer';

// 不基于dom环境的测试
beforeAll(function () {
	console.log('begin');
});
test('Layer2 renders normal', () => {
    const tree = renderer.create(
        <LayerComponent />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
test('Layer2 renders toast', () => {
    const tree = renderer.create(
        <LayerComponent type="toast" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
test('Layer2 renders loading', () => {
    const tree = renderer.create(
        <LayerComponent type="loading" />
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
afterAll(() => {
	console.log('end');
});