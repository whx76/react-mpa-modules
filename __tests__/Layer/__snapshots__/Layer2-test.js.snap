exports[`test Layer2 renders loading 1`] = `
<div>
  <div>
    <p>
      loading
    </p>
  </div>
</div>
`;

exports[`test Layer2 renders normal 1`] = `
<div>
  <div>
    <p>
      default
    </p>
  </div>
</div>
`;

exports[`test Layer2 renders toast 1`] = `
<div>
  <div>
    <p>
      toast
    </p>
  </div>
</div>
`;
